### DISCUSSION TITLE
How to post form data

### DISCUSSION THREAD

**Discussion by jwhoami** [rest-nvim/rest.nvim#259] [Q&A]
Hi,

I can post json data without problems, however the following request does not work

`POST http://localhost/api/sign-request/add
Content-Type: application/x-www-form-urlencoded
Authorization: Bearer {{API_TOKEN}}

field1=value1&field2=value2
`

The curl command outputted is as follows:

`Command: curl -sSL --compressed -X 'POST' -H 'Authorization: Bearer 1|MKpgUuh3kJr68463Csx6fHexQmdQ7YUbZT2iMiwGa7e471f5' -H 'Content-Type: application/x-www-form-urlencoded' 'http://localhost/api/sign-request/add'`

So the data is not added. 

Am i doing something wrong or the header "application/x-www-form-urlencoded" is not yet supported.

Thanks

**Comment by nicolasauler** [comment]
I have that same question!
I've been trying to post form data but can't get it to work, while if I curl it directly, it works!

**Comment by gegnew** [comment]
Tell me someone has figured out a solution to this 🙏

**Comment by boltlessengineer** [comment]
Closing this due to v3 release.
Now rest.nvim supports `x-www-form-urlencoded` content type.
